### DISCUSSION TITLE
Add additional peer to KubeSpan

### DISCUSSION THREAD

**Discussion by webwurst** [siderolabs/talos#8450] [Q&A] [answered]
Is there an easy way to connect an additional peer to the KubeSpan wireguard network? I would like to allow a wireguard connection from my laptop to Talos Kubernetes, maybe also a different edge device from my home network that can't run Kubernetes.

As a bonus: Is it then possible for the Kubernetes-API to only listen on the wireguard network?

**Comment by smira** [comment, marked as answer]
KubeSpan at the moment doesn't allow any "foreign" members.

It's a transparent layer, so you can't make Kubernetes API listen on wireguard network, but you could use a firewall and block access to the API, KubeSpan will be still allowed.
